features/git-town/configuration/configuration_initial.feature

```diff
@@ -0,0 +1,73 @@
+Feature: Initial configuration
+
+  Background:
+    Given I haven't configured Git Town yet
+
+
+  Scenario Outline: Running Git Town commands while Git Town is unconfigured
+    When I run `<COMMAND>` while allowing errors
+    Then I see
+      """
+      Git Town hasn't been configured for this repository.
+      Please run 'git town config --setup'.
+      Would you like to do that now? y/n
+      """
+
+    Examples:
+    | COMMAND            |
+    | git extract        |
+    | git hack           |
+    | git kill           |
+    | git pr             |
+    | git prune-branches |
+    | git repo           |
+    | git ship           |
+    | git sync           |
+    | git sync-fork      |
+
+
+  Scenario: Enter non-existent main branch
+    When I run `git town config --setup` and enter "nonexistent"
+    Then I see
+      """
+      Please enter the name of the main dev branch (typically 'master' or 'development'):
+
+        Error
+        There is no branch named 'nonexistent'
+
+      """
+    And Git Town is still not configured for this repository
```

Comment:
You are right, @charlierudolph. But this is such an extreme edge case that happens only once over the lifetime of the Git Town installation that we don't have to over-engineer this. A simple exit with a clean error message and the possibility to re-run the last command on the command line should be enough here.